src/WebUI/Components/KubeSummary.tsx

```diff
@@ -255,6 +283,112 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
+        />);
+    }
+
+    private _getDeployments(): JSX.Element {
+        return (<DeploymentsComponent
+            deploymentList={this.state.deploymentList || {} as V1DeploymentList}
+            replicaSetList={this.state.replicaSetList || {} as V1ReplicaSetList}
+            key={format("dc-{0}", this.state.namespace || "")}
+            onItemInvoked={this._onDeploymentItemInvoked}
+            nameFilter={this._getNameFilterKey()}
+        />);
+    }
+
+    private _onFilterApplied = (currentState: IFilterState) => {
+        this.setState({
+            workloadsFilterState: this.state.filter.getState()
+        })
+    };
+
+    private _onSvcFilterApplied = (currentState: IFilterState) => {
+        this.setState({
+            svcFilterState: this.state.svcFilter.getState()
+        })
+    };
+
+    private _getNameFilterKey(): string | undefined {
+        const filterState:IFilterState | undefined = this.state.selectedKey===workloadsPivotItemKey?this.state.workloadsFilterState:this.state.svcFilterState;
+        const filterItem: IFilterItemState | null= filterState? filterState["nameKey"]: null;
+        return filterItem? (filterItem.value as string): undefined;
+    }
+
+    private _getTypeSelection():any[] {
+        const filterState:IFilterState | undefined = this.state.selectedKey===workloadsPivotItemKey?this.state.workloadsFilterState:this.state.svcFilterState;
+        const filterItem: IFilterItemState | null= filterState? filterState["typeKey"]: null;
+        const selections: any[] = filterItem? filterItem.value : [];
+        return selections;
+    }
+
+    private _showComponent(resourceType: KubeResourceType): boolean{
+        const selections: KubeResourceType[] = this._getTypeSelection();
+        // if no selections are made, show all components
+        if(selections.length > 0) {
+            return selections.indexOf(resourceType) != -1;
+        }
+        return true;
+    }
+
+    private _pickListItems = () => {
+        return [KubeResourceType.Deployments, 
+                KubeResourceType.ReplicaSets, 
+                KubeResourceType.DaemonSets, 
+                KubeResourceType.StatefulSets, 
+                KubeResourceType.Pods];
+    };
+
+    private _listItems = (item: any)=> {
+        let name:string = "";
+        switch(item){
+            case KubeResourceType.Deployments:
+                name = Resources.DeploymentsDetailsText;
+                break;
+            case KubeResourceType.ReplicaSets:
+                name = Resources.ReplicaSetText;
+                break;
+            case KubeResourceType.DaemonSets:
+                name = Resources.DaemonSetText;
+                break;
+            case KubeResourceType.StatefulSets:
+                name = Resources.StatefulSetText;
+                break;
+            case KubeResourceType.Pods:
+                name = Resources.PodsText;
+                break;
+            default:

```

Comment:
when can this happen?

default:
                name = "Unknown";
                break;

Reply:
Unreachable code. Removed